### DISCUSSION TITLE
Kratos installation fails with helm install

### DISCUSSION THREAD

**Discussion by Digant20** [ory/kratos#2440] [Q&A] [answered]
Hi,
Iam trying to install kratos with postgresql on a kubernetes cluster using helm.

**I get the below error while installing kratos:**
`helm install -f ory-kratos-kube-config.yaml ory/kratos --generate-name --debug
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/oem/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/oem/.kube/config
install.go:178: [debug] Original chart version: ""
install.go:199: [debug] CHART PATH: /home/oem/.cache/helm/repository/kratos-0.22.2.tgz
client.go:299: [debug] Starting delete for "kratos-1651475924-job" ServiceAccount
client.go:328: [debug] serviceaccounts "kratos-1651475924-job" not found
client.go:128: [debug] creating 1 resource(s)
client.go:299: [debug] Starting delete for "kratos-1651475924-automigrate" Job
client.go:328: [debug] jobs.batch "kratos-1651475924-automigrate" not found
client.go:128: [debug] creating 1 resource(s)
client.go:528: [debug] Watching for changes to Job kratos-1651475924-automigrate with timeout of 5m0s
client.go:556: [debug] Add/Modify event for kratos-1651475924-automigrate: ADDED
client.go:595: [debug] kratos-1651475924-automigrate: Jobs active: 0, jobs failed: 0, jobs succeeded: 0
client.go:556: [debug] Add/Modify event for kratos-1651475924-automigrate: MODIFIED
client.go:595: [debug] kratos-1651475924-automigrate: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
Error: INSTALLATION FAILED: failed pre-install: timed out waiting for the condition
helm.go:88: [debug] failed pre-install: timed out waiting for the condition
INSTALLATION FAILED
main.newInstallCmd.func2
        helm.sh/helm/v3/cmd/helm/install.go:127
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371`

`$ k get pods
NAME                                     READY   STATUS                       RESTARTS   AGE
kratos-1651475924-automigrate--1-vlvk6   0/1     CreateContainerConfigError   0          17m`

**here's my kratos-config.yaml: i was not able to upload the yaml file so I am uploading the config in a txt file**
[kratos-config.txt](https://github.com/ory/kratos/files/8601493/kratos-config.txt)

Can someone please look into my kratos config if it looks fine. Thank you!

**Comment by vinckr** [comment, marked as answer]
Hello @Digant20 
I am not sure if .txt will work for the config, why is using yaml not an option?

Very hard to tell what is going wrong without having access to your infra. 

I suggest you do a review of your general Kubernetes setup and make sure everything (apart from Ory) is working correctly. If it still fails, feel free to [join our chat](https://slack.ory.sh/) and look in the `#selfhosted` channel, maybe this problem came up before.

**Reply by Digant20** [reply]
Hi @vinckr 
Thank you for a quick response.
I am using an yaml configuration but to upload the yaml config file is not supported here on github discussions, so I created a .txt file with my kratos yaml configuration and uploaded in the question here on guthub for your reference.

I am installing kratos on a fresh digitalocean kubernetes cluster.

If you can kindly verify my kratos configuration [https://github.com/ory/kratos/files/8601493/kratos-config.txt](url) that would be great. Thank you.

**Reply by vinckr** [reply]
I see - you can also use [gist](https://gist.github.com/) for this purpose, there you can upload any fileformat. 
Your config looks OK (by the way you leak some secrets and so on in the txt, probably no biggie since its local development(?) but best redact them in the future 😉).
Also take a look at the [quickstart config](https://raw.githubusercontent.com/ory/kratos/master/contrib/quickstart/kratos/email-password/kratos.yml) and [one of mine](https://gist.github.com/vinckr/3d8d36ff5584654bf4250c2bec24c224). 
I can't comment on the kubernetes part of the config.

Check out this [deployment guide for Kratos](https://docs-qzsvmlhe5-ory.vercel.app/docs/kratos/guides/going-to-production-with-nginx-and-digitalocean) - it is still in review phase so take it with a grain of salt, but we will add the reviewed/edited version to the docs soon.

**Reply by vinckr** [reply]
Hey @Digant20, just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
